zipline/assets/assets.py

```diff
@@ -261,28 +261,50 @@ def _retrieve_futures_contract(self, sid):
         self._future_cache[sid] = future
         return future
 
-    def lookup_symbol_resolve_multiple(self, symbol, as_of_date=None):
+    def lookup_symbol(self, symbol, as_of_date, fuzzy=False):
         """
-        Return matching Asset of name symbol in database.
+        Return matching Equity of name symbol in database.
 
-        If multiple Assets are found and as_of_date is not set,
+        If multiple Equities are found and as_of_date is not set,
         raises MultipleSymbolsFound.
 
-        If no Asset was active at as_of_date raises SymbolNotFound.
+        If no Equity was active at as_of_date raises SymbolNotFound.
         """
+
+        # Format inputs
+        symbol = symbol.upper()
         if as_of_date is not None:
             as_of_date = pd.Timestamp(normalize_date(as_of_date))
 
+        company_symbol, share_class_symbol, fuzzy_symbol = \
```

Comment:
Don't we really only want to do this if either we have fuzzy or if we have the first failure case? Because in the "normal" case, we're assuming the symbol is just a single complete, and then if multiple THAT'S when we try to split on delimiter.